hazelcast/src/main/java/com/hazelcast/cardinality/impl/hyperloglog/impl/HyperLogLogImpl.java

```diff
@@ -95,11 +106,20 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeObject(encoder);
+
+        if (out.getVersion().isGreaterOrEqual(Versions.V3_10)) {
+            out.writeInt(m);
```

Comment:
Why do you need to transfer `m`? Is this in case the merge policy uses the merging instance of `HyperLogLogImpl`?

Reply:
That was a bug, we never transferred it, and its needed for the impl. Nothing to do with merging.